src/app/store/courses/courses.effects.ts

```diff
@@ -1,8 +1,118 @@
-import { Injectable } from '@angular/core';
+import { Injectable } from "@angular/core";
+import { Actions, ofType, createEffect } from "@ngrx/effects";
+import { mergeMap, map, catchError } from "rxjs/operators";
+import { EMPTY } from "rxjs";
+
+import { CoursesService } from "@app/services/courses.service";
+import {
+  requestAllCourses,
+  requestAllCoursesSuccess,
+  requestAllCoursesFail,
+  requestSingleCourse,
+  requestSingleCourseSuccess,
+  requestSingleCourseFail,
+  requestDeleteCourse,
+  requestDeleteCourseSuccess,
+  requestDeleteCourseFail,
+  requestEditCourse,
+  requestEditCourseSuccess,
+  requestEditCourseFail,
+  requestCreateCourse,
+  requestCreateCourseSuccess,
+  requestCreateCourseFail,
+} from "./courses.actions";
 
 @Injectable()
 export class CoursesEffects {
-    constructor() {}
+  getAll$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(requestAllCourses),
+      mergeMap(() =>
+        this.courseService.getAll().pipe(
+          map((res) => requestAllCoursesSuccess({ courses: res.result })),
+          catchError((e) => {
+            requestAllCoursesFail({ error: e });
```

Comment:
you're not dispatching error action here, so the store will not be updated with an error